Mathlib/Geometry/Synthetic/Avigad/Axioms.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2023 André Hernandez-Espiet. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: André Hernandez-Espiet
+-/
+import Mathlib.Data.Real.Basic
+import Mathlib.Data.Set.Finite
+
+/-!
+Axioms of synthetic geometry
+-/
+
+/-! Universes for points lines and circles-/
+universe u
+
+/-- `IncidenceGeometry` represents geometry in the Euclidean sense, with primitives for points
+lines and circles-/
+class IncidenceGeometry where
+/--Points in the plane-/
+Point : Type u
+/--Lines in the plane-/
+Line : Type u
+/--Circles in the plane-/
+Circle : Type u
+
+/--A Point being on a line-/
+OnLine : Point → Line → Prop
+/--Two points being on the sameside of a line-/
+SameSide : Point → Point → Line → Prop
+/--Three points being in a row-/
+B : Point → Point → Point → Prop
+/--A point being the center of a center-/
+CenterCircle : Point → Circle → Prop
+/--A point being on a circle-/
+OnCircle : Point → Circle → Prop
+/--A point being inside a circle-/
+InCircle : Point → Circle → Prop
+/--That two lines intersect-/
+LinesInter : Line → Line → Prop
+/--A line and a circle intersect-/
+LineCircleInter : Line → Circle → Prop
+/--Two circles intersect-/
+CirclesInter : Circle → Circle → Prop
+/--The distance between two points-/
+length : Point → Point → ℝ
+/--The angle made between three points-/
+angle : Point → Point → Point → ℝ
+/--The constant dedicated to the right angle-/
+rightangle : ℝ
+/--The area made by three points-/
+area : Point → Point → Point → ℝ
+
+/--From a set of points getting one additional one-/
+more_pts : ∀ (S : Set Point), S.Finite → ∃ a, a ∉ S
```

Comment:
Would it make more sense to instead assume that `Point` is infinite?

Reply:
How would you suggest going about this? I am not really sure how to do it.

Reply:
```suggestion
[more_pts : Infinite Point]
```

Reply:
Interesting... How would I use this condition to get the same effect as the original? Previously I could write: 
`rcases more_pts ∅ Set.finite_empty with ⟨a, -⟩` to get a point, for example.

Reply:
Apparently we're missing a `Set.Finite` version of [Infinite.exists_not_mem_finset](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Card.html#Infinite.exists_not_mem_finset)?